### ISSUE TITLE
Automate discovery of CVE work items communicate to partners

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/source-build#3776]
Relates to https://github.com/dotnet/source-build/issues/3196, proposed by https://github.com/dotnet/source-build/issues/3196#issuecomment-1840282679

As part of the source-build release process, the source-build team communicates to partners what exact CVEs have been resolved by a specific release. This is achieved by linking the AzDO work items for each of the CVEs in a dedicated Wiki page that is then shared with partners.

Currently the links are procured manually by a member of source-build team. With the automation of the source-build partner notifications in https://github.com/dotnet/source-build/issues/3196, it would be helpful if the source-build release pipeline could automatically pick-up the work items it needs based on some pre-defined logic. This logic could either be some tag that is present on all relevant work items or it could be a list of the CVEs themselves (such a list is already part of the release `config.json`)

We should identify the best way to achieve this automation and incorporate it into the source-build partners notification process. If needed, we should contact the .NET release team and discuss with them the best ways to achieve this.

**Comment by oleksandr-didyk** [comment]
Release config (`config.json`) we retrieve from the staging pipeline contains a list of CVE's that are fixed by the current servicing release. Example for `6.0.26`:
```json
  "CVE_List": [
    {
      "cve_id": "CVE-2024-0056 CVE-2024-0057 CVE-2024-0058"
    }
  ],
```

As for finding the corresponding work items in `dotnet-security-partners`, we could simply fetch all of the work items in a specific project through [REST API](https://learn.microsoft.com/en-us/rest/api/azure/devops/wit/work-items/list?view=azure-devops-rest-7.1&tabs=HTTP) and then filter them by name to get the work items we should include in the Wiki or just query them directly through [WIQL](https://learn.microsoft.com/en-us/rest/api/azure/devops/wit/wiql/query-by-wiql?view=azure-devops-rest-5.1&tabs=HTTP#request-body). To make the interaction with AzDO a bit less painful we could also try utilizing the C# [client library](https://learn.microsoft.com/en-us/azure/devops/integrate/get-started/client-libraries/samples?view=azure-devops).

**Comment by oleksandr-didyk** [comment]
I skimmed through the work items in the repo and could not find a work item that would be tagged with a release. As an example here are the tags for one of the CVE's mentioned in the example above:
<img width="193" alt="image" src="https://github.com/dotnet/source-build/assets/106967057/6fba7665-4b8b-49f3-83b2-94d8284fa31a">

Additionally, the work item description also does not state anything about the planned release that will address it

**Comment by mthalman** [comment]
Right, the `2024.01.B` tag is what would be searched for in the case of the January release.

**Comment by oleksandr-didyk** [comment]
PR - https://dev.azure.com/dnceng/internal/_git/dotnet-release/pullrequest/37338

**Comment by oleksandr-didyk** [comment]
PR merged, closing
